runtime/lua/vim/keymap.lua

```diff
@@ -60,22 +59,9 @@ function keymap.set(mode, lhs, rhs, opts)
   local is_rhs_luaref = type(rhs) == 'function'
   mode = type(mode) == 'string' and { mode } or mode
 
-  if is_rhs_luaref and opts.expr then
-    local user_rhs = rhs
-    rhs = function()
-      local res = user_rhs()
-      if res == nil then
-        -- TODO(lewis6991): Handle this in C?
-        return ''
```

Comment:
This revert the feature introduced in #17842

Reply:
I tried to understand why this is there, but I couldn't really tell the difference? I see now that Lewis mentions that it used to throw an error, but this seems to no longer be the case. Is it possible that something changed in the implementation and this is no longer relevant?

I could do something like this, but it would be nice to explain in the comment why is it here:
```lua
  if (p == NULL) {
    return mp->m_replace_keycodes ? (char_u *)xstrdup("") : NULL;
  }
```

Reply:
@lewis6991

Reply:
Oh `false` is removed. Then this indeed doesn't change behavior.